### PR TITLE
Issue #70 - Detect kotlin coroutine calls.

### PR DESCRIPTION
**Pull request by croudet**

### CODE REVIEW

File: micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java [hunk 1]
```diff
@@ -75,86 +79,123 @@ protected TimedInterceptor(MeterRegistry meterRegistry) {
...
+            // not suspended
+            stopTimed(
+                    metricName,
+                    sample,
+                    "none",
+                    metadata);
+        } else {
+            // last argument
+            final String[] argumentNames = context.getArgumentNames();
+            final MutableArgumentValue<?> argument = context.getParameters().get(argumentNames[argumentNames.length - 1]);
```
Comment (graemerocher) [inline micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java:179]: `context.getParameters()` is relatively expensive, and you don't need the `MutableArgument` value for these case. Use `getParameterValues()[argumentNames.length - 1]` instead which will provide better performance

File: micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java [hunk 2]
```diff
@@ -75,86 +79,123 @@ protected TimedInterceptor(MeterRegistry meterRegistry) {
...
+                            sample,
+                            exception == null ? "none" : exception.getClass().getSimpleName(),
+                            metadata)
+                );
+            }
+        }
+        return result;
+    }
+
+    private static boolean isKotlinContinuation(MethodInvocationContext<Object, Object> context) {
```
Comment (graemerocher) [inline micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java:194]: I believe for this you can just call `context.getExecutableMethod().isSuspend()`

**Comment by graemerocher** [comment]
Couple of changes, thanks for the contribution!

File: micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java (hunk 1 above)
Comment (croudet) [reply micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java]: Done

File: micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java (hunk 2 above)
Comment (croudet) [reply micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java]: Done

**Comment by croudet** [comment]
Are we sure that the last argument will always be of type `CustomContinuation`?

**Comment by graemerocher** [comment]
Thanks for the contribution!

**Pull request merged**